### ISSUE TITLE
using !expr in chunk YAML

### ISSUE THREAD

**Issue by BradyAJohnston** [quarto-dev/quarto-cli#547]
I'm trying to test out using the `!expr` in the chunk YAML to include calculated results for figure legends. In the documentation it says it is [possible:](https://quarto.org/docs/tools/rstudio.html#knitr-engine)
```
#| fig-cap: !expr paste("Air", "Quality")
```

But when in attempting to use it, I am so far met with errors both when attempting to render and RStudio detecting errors:
````r
---
title: "test"
---

```{r}
#| label: fig-test
#| fig-cap: !expr paste("Air", "Quality")
plot(mtcars)
```
````

```
ERROR: YAMLError: unknown tag !<!expr> at line 1, column 39:
     ... p: !expr paste("Air", "Quality")
                                         ^
```

<img src="https://user-images.githubusercontent.com/36021261/161966409-9233d7e2-2fae-4db3-b1d9-4217e9a8161e.png">

**Comment by cscheid** [comment]
We added a fix for this in 32f0c0edbcaa634e3fde788c874acab98e4e31f4. Let me know if this doesn't work!

**Comment by jjallaire** [comment]
Note that currently these expressions are evaluated at the beginning of knit (there is a PR into knitr to ensure they are evaluated at cell execution time: https://github.com/yihui/knitr/pull/2120)
